intro-to-functions/practice-intro-to-functions.md

```diff
@@ -0,0 +1,935 @@
+    return entire_song
+```
+
+Technically, we could re-orer the assignments of `line_1`, `line_2`, `line_3`, and `line_4` to any combination we want. However, it's ideal to have them ordered because of readability and coherence to all programmers. :)
+
+##### !end-explanation
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+### Reading Code
+
+<!-- Question 15 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: multiple-choice
+* id: 2rhUAe
+* title: Intro to Functions
+##### !question
+
+Read through this code. Assume that `get_season()` and `get_day_of_week()` are valid functions. What does this function do?
+
+```python
+def get_daily_special(todays_date):
+    special = ""
+
+    current_season = get_season(todays_date)
+
+    if current_season == "Spring":
+        special = "Eggplants"
+    elif current_season == "Summer":
+        special = "Blueberries"
+    elif current_season == "Fall":
+        special = "Sweet Potatoes"
+    else:
+        special = "Oranges"
+
+    day_of_week = get_day_of_week(todays_date)
+
+    if day_of_week == "Saturday" or day_of_week == "Sunday":
+        special = f"Weekend {special}"
+
+    return special
+```
+
+##### !end-question
+##### !options
+
+* Sets all of the season and weekend specials
+* Determines what season and day of the week it is
+* Determines the special based on today's season and day of the week
+* Concatenates today's date with the season and day of the week
+
+##### !end-options
+##### !answer
+
+* Determines the special based on today's season and day of the week
+
+##### !end-answer
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+### Debugging Functions
+
+<!-- Question 16 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: multiple-choice
+* id: Rxvbnm
+* title: Intro to Functions
+##### !question
+
+Assuming that `get_season()` and `get_day_of_week()` are valid functions, given this code...
+
+```python
+def get_daily_special(todays_date):
+special = ""
+
+    current_season = get_season(todays_date)
+
+    if current_season == "Spring":
+        special = "Eggplants"
+    elif current_season == "Summer":
+        special = "Blueberres"
+    elif current_season == "Fall":
+        special = "Sweet Potatoes"
+    else:
+        special = "Oranges"
+
+    day_of_week = get_day_of_week(todays_date)
+
+    if day_of_week == "Saturday" or day_of_week == "Sunday":
+        special = f"Weekend {special}"
+
+    return special
+```
+
+There is a syntax or runtime error in this code. What is the cause of this error?
+
+##### !end-question
+##### !options
+
+* Logical Error
+* Incorrect parameters
+* Incorrect Indentation
+* Missing colon
+* Missing parentheses
+* Reads a variable that has no value
+* Mispelling
+
+##### !end-options
+##### !answer
+
+* Incorrect Indentation
+
+##### !end-answer
+##### !hint
+
+If there are typos within a string literal, it may be wrong spelling, but it will not cause a syntax or runtime error.
```

Comment:
i have very weirdly personal reasons to fight for this middle clause LOL, let me know if you want me to defend it in words (otherwise i'll defend it in inaction LOL)

Reply:
Well now I'm curious...